### PR TITLE
fix: always use parameter UTF-8 fallback

### PR DESCRIPTION
**Pull request by muggenhor**
Even when the filesystem encoding and sys.argv encoding are the same we
should still fall back to trying UTF-8 for decoding command line
arguments.

### CODE REVIEW

**Comment by davidism** [comment]
This *seems* ok, but I can't tell what it might affect. Can you add a test case?

File: click/types.py [hunk 1]
```diff
@@ -119,6 +119,8 @@ def convert(self, value, param, ctx):
                         value = value.decode(fs_enc)
                     except UnicodeError:
                         value = value.decode('utf-8', 'replace')
+                else:
+                    value = value.decode('utf-8', 'replace')
```
Comment (hg-pyun) [inline click/types.py]: Please, Add TC :)

Comment (muggenhor) [reply click/types.py]: Working on it. I'll aim to push a test case next week.

**Comment by muggenhor** [comment]
I've made the test case commit come first. That way it's easier to verify that it fails without the fix commit.

**Comment by davidism** [comment]
Just leaving a note for myself in case I need to revisit this later. It seems like `.decode("utf-8", "replace")` is already used as a fallback, it just wasn't applied consistently. Since this seems to improve consistency, I'm merging it.

Thanks for working on this!

**Pull request merged**